### ISSUE TITLE
Dynamic reconfiguration of clustered resource pool does not work

### ISSUE THREAD

**Issue by kedar031** [ehcache/ehcache3#1582]
User is not prevented to dynamically update clustered resource (no unsupported exception occurs), however it does not take effect (although cache configuration shows the newly updated value).
No validation of pool resources occurs, able to update size of clustered resource pool to any integer value > 0 

An example as below
````
Cache cache1 = cacheManager.getCache("clustered-cache1", Long.class, String.class);
    ResourcePools pools = ResourcePoolsBuilder.newResourcePoolsBuilder()
        .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 999999999, MemoryUnit.MB)).build();
cache1.getRuntimeConfiguration().updateResourcePools(pools);
````

**Comment by ljacomet** [comment]
To be clear, we will not support resizing of clustered tiers for now - but this should properly give an exception to the user.

**Comment by ljacomet** [comment]
needs to be fixed on `release/3.1` and `master`
